tests/subsys/logging/log_blocking/REAME.md

```diff
@@ -0,0 +1,97 @@
+# Blocking the Log Processing Thread
+
+## Overview
+
+When the core log buffer becomes full, the logging subsystem can be configured to
+
+* Drop older log messages with `CONFIG_LOG_MODE_OVERFLOW=y` (**default**)
+* Drop newer log messages with `CONFIG_LOG_MODE_OVERFLOW=n`, or
+* Drop no log messages at all with `CONFIG_LOG_BLOCK_IN_THREAD=y`, `CONFIG_LOG_BLOCK_IN_THREAD_TIMEOUT_MS=-1`.
+
+In the last configuration, the log processing thread will block until space
+becomes available again in the core log buffer.
+
+> Warning ⚠️: Blocking the log processing thread is generally not recommended
+> and should only be attempted in advanced use cases.
+
+## Logging and Flow Rates
+
+There are roughly 4 scenarios we care about testing with
+`CONFIG_LOG_BLOCK_IN_THREAD`, and they can all be characterized by comparing
+log message flow rates. Typically, one would describe log message flow rates
+with units such as `[msg/s]`. However, in the table below, we are mainly
+concerned with the ratio of the Output Rate to the Input Rate, and in that
+case, the units themselves cancel-out. In the table we assume there exists an
+`N` such that `N > 1`.
+
+| Name           | Input Rate | Output Rate | Rate |
+|----------------|------------|-------------|------|
+| Input-Limited  | 1          | N           | 1    |
+| Matched        | 1          | 1           | 1    |
+| Output-Limited | 1          | 1/N         | 1/N  |
+| Stalled        | 0          | 0           | 0    |
+
+The resultant _Rate_ is always `Rate = MIN(Input Rate, Output Rate)`.
```

Comment:
Heh... maybe a follow up PR (feel free)? It might not be worth trying to get another round of approvals.

GitHub has a few shortcomings: 
* previous approvals are invalidated (maybe this isn't the case when adding subsequent fixup commits?)
* cannot explicitly hand off a PR to someone else (afaik, e.g. #65769)

If it were possible to not need to get re-approvals if only fixup commits were added, then that should probably be the norm, and then make the default merge button "squash and merge".